Because the authors are expecting the extracted sample to be used for next-generation sequencing, the protocol includes the WGA. But I don't think our protocol needs to include that step, since it will be imaged on your device. To make matters worse, they wrote the methods of their article with "Whole Genome Amplification" as the second subsection, even though it is performed after the RSE.

So what's needed is 

1. The oligonucleotides
2. Something like the H-reaction buffer which they write "contains polymerase, dNTPs, and biotinylated dNTPs".  Unfortunately, that's one of the buffers they sell, and they are not more specific about it. I was able to find, however, another publication from before they commercialised it, and they used a buffer from Qiagen https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.0020212

So I think you need the following:

1. The primers.
2. a dNTP mix for primer extension
3. A biotinylated dNTP for capture
4. magnetic strepatvidin beads.
5. A hybridisation buffer containing a thermostable DNA polymerase (Qiagen may have one).
